compiler/ballerina-lang/src/main/java/io/ballerina/projects/SemanticVersion.java

```diff
@@ -132,12 +132,11 @@ public VersionCompatibilityResult compareTo(SemanticVersion other) {
             return VersionCompatibilityResult.EQUAL;
         }
 
-        // Eliminate initial versions, pre-release and build metadata
-        if (!this.isStable() || !other.isStable()) {
+        if (this.major() != other.major()) {
             return VersionCompatibilityResult.INCOMPATIBLE;
         }
 
-        if (this.major() != other.major()) {
+        if (this.isInitialVersion()) {
```

Comment:
This means per-relases of same initial versions are also incompatible. Shall we add some text cases for that behavior.

Reply:
Ack will add

Reply:
Added with https://github.com/ballerina-platform/ballerina-lang/pull/30846